docs/advanced/payments/payment-workflows-specificities.mdx

```diff
@@ -158,3 +165,138 @@ const getAdditionalActionComponent = (paymentCode, paymentAdditionalData) => {
 
 export default getAdditionalActionComponent;
 ```
+
+## Redirect Before Order
+
+:::warning
+
+This workflow is only supported for Magento2, please
+[contact us](mailto:contact@front-commerce.com) if you need this workflow with
+another backend server
+
+:::
+
+A redirect before order workflow allows to interact with payment providers
+handling the complete payment workflow on their side.
+
+Front-Commerce will only redirect to the payment provider and ensure the order
+creation afterwards.
+
+### Server changes
+
+1. Register the payment method payment processor with
+   `registerRedirectBeforeOrderPaymentMethod` module’s
+   [`contextEnhancer`](/docs/reference/graphql-module-definition#contextenhancer-optional)
+
+Replace the module's contextEnhancer with the following
+
+```js
+import { getCurrentShopConfig } from "server/core/config/getShopConfig";
+
+export default {
+  namespace: "Payments/PWAy",
+  dependencies: ["Magento2/Checkout", "Front-Commerce/Payment"],
+  contextEnhancer: ({ loaders }) => {
+    const shopConfig = getCurrentShopConfig(req.config);
+    const METHOD_CODE = "pway_awesomecheckout";
+    const METHOD_TITLE = "PWAy";
+
+    // [...] initialization here
+    const loader = new MyPaymentLoader(shopConfig, METHOD_CODE, ...);
+
+    loaders.Payment.registerRedirectBeforeOrderPaymentMethod(
+      METHOD_CODE,
+      METHOD_TITLE,
+      loader,
+    );
+
+    return {}; // you may export loaders here in case the payment provides custom Queries (to fetch a payment token for instance)
+  },
+};
+```
+
+2. Implement the three **required** methods in the loader
+
+```js
+export default class MyPaymentLoader {
+  constructor(shopConfig, methodCode, ...) {
+    ...
+
+    this.callbackUrl = `${shopConfig.url}/checkout/payment/${methodCode}`;
+  }
+
+  mapCartPaymentToRedirect({ paymentData /* TODO ? */, cart }) {
+    // reminder : the callbackUrl to FC is calculated in this.callbackUrl (defined in the constructor)
```

Comment:
I don't understand this comment. `this.callbackUrl` is not used anywhere. Is it supposed to be used in `value` below ?